### ISSUE TITLE
Device frequently unavailable

### ISSUE THREAD

**Issue by swratten** [nikrolls/homeassistant-goldair-climate#36]
Hi there,
I've managed to get this running on my GPCV330.
I've noticed the logs full of the device becoming unavailable.
I'm on my phone so I'll attach some screenshots of what I can see.
The signal strength seems to be around -56dbm so should be good there.
My automation for turning off my son's heater fired this morning but didn't seem to stick as it conveniently became unavailable at the time...
Is this just a bad unit or something in the polling?
```
020-06-28 04:00:31 ERROR (Thread-10) [custom_components.goldair_climate.device] Failed to update device state
```
![Screenshot_20200628-082657](https://user-images.githubusercontent.com/4234090/85932597-71a63a00-b921-11ea-8803-b090bcc9cb12.png)
![Screenshot_20200628-084502](https://user-images.githubusercontent.com/4234090/85932601-78cd4800-b921-11ea-835d-dc0685d1f89b.png)

**Comment by bedazbedaz** [comment]
I have 4 of these heaters, and I'm still working through what's going on with one of them as I've had some issues with reliability.

How much are you using the app? I have noticed that any time the app is open (even in background on android) it seems to mess up the states. Once I close the app, the control seems to pass back to the homeassistant integration after between 5 and 30 seconds or so. I haven't investigated much further, but I thought I'd pass it on as it's the first lead I've had (and I'm sure it's in the documentation somewhere..)

**Comment by swratten** [comment]
> I have 4 of these heaters, and I'm still working through what's going on with one of them as I've had some issues with reliability.
> 
> How much are you using the app? I have noticed that any time the app is open (even in background on android) it seems to mess up the states. Once I close the app, the control seems to pass back to the homeassistant integration after between 5 and 30 seconds or so. I haven't investigated much further, but I thought I'd pass it on as it's the first lead I've had (and I'm sure it's in the documentation somewhere..)

If you are referring to the Goldair Android application, I used it to register the heater and get the encryption key and haven't used it since (and cleared it from background tasks). 
This happens constantly when I am not at home as well so the Android application couldn't possibly be locking the device up.

**Comment by bedazbedaz** [comment]
Yes I was referring to the android app, that could have been clearer sorry. 

Oh well, good luck - mine have been stable for a few days now, but I don't have anything useful to add. If it starts happening again, I'll try to contribute some logs

**Comment by make-all** [comment]
I have a number of devices, most are on Tuya Protocol 3.1 still, and are very stable, but one is on Tuya protocol 3.3, and has this behavior.  Every now and then, a response is not recognized by the pytuya library, and in response to the error, the homeassistant-goldair-climate component switches protocol to 3.1 and retries, then switches back to 3.3 and retries, but the heater remains unresponsive for some time so it takes a lot of these retries to start working again.

I'm not sure whether the unresponsiveness is caused by the initial condition that causes the heater to send back the reply that pytuya doesn't recognize, or if it is the protocol 3.1 retry. If the latter, then a change to be less eager to change protocols when a device has already been working on one protocol might help.
